core/src/com/unciv/logic/civilization/CivilizationInfo.kt

```diff
@@ -591,5 +595,15 @@ class CivilizationInfo {
         }
     }
 
+    // Can't get ruins bonuses several times in a row - only once every 3 ruins
+    fun recordPickRuinBonus(bonus: AncientRuins.RuinBonus) {
+        blockedRuinsBonus[bonus] = 2
+    }
+
+    fun updateAndGetBlockedRuinsBonus(): Set<AncientRuins.RuinBonus> {
+        blockedRuinsBonus.forEach { (s, i) -> blockedRuinsBonus[s] = blockedRuinsBonus[s]!! - 1}
```

Comment:
This line is unclear - if you iterated on the keys, named the string parameter, and checked for 0 values, you couls make this whole function much more readable